packages/client/hmi-client/src/components/model/tera-model.vue

```diff
@@ -0,0 +1,168 @@
+<template>
+	<!--main is temporary just for css purposes-->
+	<main>
+		<tera-asset
+			v-if="model"
+			:name="model.name"
+			:feature-config="featureConfig"
+			:is-naming-asset="isNamingModel"
+			:stretch-content="modelView === ModelView.MODEL"
+		>
+			<template #name-input>
+				<!--@keyup.enter="updateModelName" this is being reworked in rename-dataset branch-->
+				<InputText
+					v-if="isNamingModel"
+					v-model.lazy="newModelName"
+					placeholder="Title of new model"
+				/>
+			</template>
+			<template #edit-buttons>
+				<span class="p-buttonset">
+					<Button
+						class="p-button-secondary p-button-sm"
+						label="Description"
+						icon="pi pi-list"
+						@click="modelView = ModelView.DESCRIPTION"
+						:active="modelView === ModelView.DESCRIPTION"
+					/>
+					<Button
+						class="p-button-secondary p-button-sm"
+						label="Model"
+						icon="pi pi-file"
+						@click="modelView = ModelView.MODEL"
+						:active="modelView === ModelView.MODEL"
+					/>
+					<Button
+						class="p-button-secondary p-button-sm"
+						label="Transform"
+						icon="pi pi-sync"
+						@click="modelView = ModelView.NOTEBOOK"
+						:active="modelView === ModelView.NOTEBOOK"
+					/>
+				</span>
+				<template v-if="!featureConfig.isPreview">
+					<!--Disable this until rename-dataset is merged-->
+					<Button
+						disabled
+						icon="pi pi-ellipsis-v"
+						class="p-button-icon-only p-button-text p-button-rounded"
+						@click="toggleOptionsMenu"
+					/>
+					<Menu ref="optionsMenu" :model="optionsMenuItems" :popup="true" />
+				</template>
+			</template>
+			<tera-model-description v-if="modelView === ModelView.DESCRIPTION" :model="model" />
+			<Accordion v-else-if="modelView === ModelView.MODEL" multiple :active-index="[0, 1, 2, 3]">
+				<AccordionTab header="Model diagram">
+					<tera-model-diagram
+						:model="model"
+						:is-editable="!featureConfig.isPreview"
+						@update-model="updateModelContent"
+					/>
+				</AccordionTab>
+				<AccordionTab header="Model equations"> </AccordionTab>
+				<AccordionTab header="Model observables"></AccordionTab>
+				<AccordionTab header="Model configurations"></AccordionTab>
+			</Accordion>
```

Comment:
This should be a component itself.
Not on this PR, but the next one.

Reply:
That makes sense thanks